From 3320dc3c500133beace7fd0c75841cd85dd0ea22 Mon Sep 17 00:00:00 2001
From: "johan.bohlin" <johan.bohlin@intel.com>
Date: Mon, 17 Nov 2014 14:25:27 +0100
Subject: [PATCH 3/7] ASoC: codecs: rt5640 remove deprecated dapm muxes

Remove deprecated INL/INR dapm muxes that are setting
reserved registers which should not be changed by the
codec driver.

Issue: GMINL-3484

Change-Id: I18e225ab095577abb3729de8607ecf7212146a74
Signed-off-by: johan.bohlin <johan.bohlin@intel.com>
---
 sound/soc/codecs/rt5640.c |   20 --------------------
 sound/soc/codecs/rt5640.h |    2 --
 2 files changed, 22 deletions(-)

diff --git a/sound/soc/codecs/rt5640.c b/sound/soc/codecs/rt5640.c
index 3db173a..6a53b57 100644
--- a/sound/soc/codecs/rt5640.c
+++ b/sound/soc/codecs/rt5640.c
@@ -1254,23 +1254,6 @@ static const struct snd_kcontrol_new rt5640_mono_mix[] = {
 			RT5640_M_BST1_MM_SFT, 1, 1),
 };
 
-/* INL/R source */
-static const char * const rt5640_inl_src[] = { "IN2P", "MonoP" };
-
-static const SOC_ENUM_SINGLE_DECL(rt5640_inl_enum, RT5640_INL_INR_VOL,
-				  RT5640_INL_SEL_SFT, rt5640_inl_src);
-
-static const struct snd_kcontrol_new rt5640_inl_mux =
-SOC_DAPM_ENUM("INL source", rt5640_inl_enum);
-
-static const char * const rt5640_inr_src[] = { "IN2N", "MonoN" };
-
-static const SOC_ENUM_SINGLE_DECL(rt5640_inr_enum, RT5640_INL_INR_VOL,
-				  RT5640_INR_SEL_SFT, rt5640_inr_src);
-
-static const struct snd_kcontrol_new rt5640_inr_mux =
-SOC_DAPM_ENUM("INR source", rt5640_inr_enum);
-
 /* Stereo ADC source */
 static const char * const rt5640_stereo_adc1_src[] = { "DIG MIX", "ADC" };
 
@@ -1989,9 +1972,6 @@ static const struct snd_soc_dapm_widget rt5640_dapm_widgets[] = {
 			 RT5640_PWR_IN_L_BIT, 0, NULL, 0),
 	SND_SOC_DAPM_PGA("INR VOL", RT5640_PWR_VOL,
 			 RT5640_PWR_IN_R_BIT, 0, NULL, 0),
-	/* IN Mux */
-	SND_SOC_DAPM_MUX("INL Mux", SND_SOC_NOPM, 0, 0, &rt5640_inl_mux),
-	SND_SOC_DAPM_MUX("INR Mux", SND_SOC_NOPM, 0, 0, &rt5640_inr_mux),
 	/* REC Mixer */
 	SND_SOC_DAPM_MIXER("RECMIXL", RT5640_PWR_MIXER, RT5640_PWR_RM_L_BIT, 0,
 			   rt5640_rec_l_mix, ARRAY_SIZE(rt5640_rec_l_mix)),
diff --git a/sound/soc/codecs/rt5640.h b/sound/soc/codecs/rt5640.h
index 53c9e5b..932cc66 100644
--- a/sound/soc/codecs/rt5640.h
+++ b/sound/soc/codecs/rt5640.h
@@ -224,13 +224,11 @@
 
 /* INL and INR Volume Control (0x0f) */
 #define RT5640_INL_SEL_MASK			(0x1 << 15)
-#define RT5640_INL_SEL_SFT			15
 #define RT5640_INL_SEL_IN4P			(0x0 << 15)
 #define RT5640_INL_SEL_MONOP			(0x1 << 15)
 #define RT5640_INL_VOL_MASK			(0x1f << 8)
 #define RT5640_INL_VOL_SFT			8
 #define RT5640_INR_SEL_MASK			(0x1 << 7)
-#define RT5640_INR_SEL_SFT			7
 #define RT5640_INR_SEL_IN4N			(0x0 << 7)
 #define RT5640_INR_SEL_MONON			(0x1 << 7)
 #define RT5640_INR_VOL_MASK			(0x1f)
-- 
1.7.9.5

